.github/workflows/compile-queries.yml

```diff
@@ -18,12 +18,18 @@ jobs:
           fetch-depth: 0
       # calculate the merge-base with main, in a way that works both on PRs and pushes to main. 
       - name: Calculate merge-base
-        id: merge-base
+        if: ${{ github.event_name == 'pull_request' }}
         env:
           BASE_BRANCH: ${{ github.base_ref }}
         run: |
           MERGE_BASE=$(git merge-base --fork-point origin/$BASE_BRANCH)
           echo "merge-base=$MERGE_BASE" >> $GITHUB_ENV
+      - name: Calculate merge-base - main
+        if: ${{ github.event_name != 'pull_request' }}
+        # using github.sha instead, since we're on main
```

Comment:
```suggestion
      - name: Calculate merge-base - branch
        if: ${{ github.event_name != 'pull_request' }}
        # using github.sha instead, since we're directly on a branch, and not in a PR
```